### DISCUSSION TITLE
Importing the OSDP library into the project

### DISCUSSION THREAD

**Discussion by mbrazdil** [goToMain/libosdp#180] [Q&A] [answered]
Hello,

I am working on a project in the MCUXpresso IDE (C language) and I have the task to import the OSDP library into the project. I've read everything, but nowhere can I find a procedure to add this library to the project. Which parts of the library exactly should I copy into the project? Or do I have to compile the library? 
I tried to copy the whole kinhovna folder to the project, but it gives me errors with missing header files e.g. osdp_export.h.

**Comment by sidcha** [comment, marked as answer]
Hi @mbrazdil, The typical work flow is to compile LibOSDP to target architecture and then link it to your project. I'm not familiar with this particular IDE, but I'm fairly confident this can be done; and it is the best way to add new platform support (please send PRs if you managed to do this).

The other (hacky) way is to import the source files into your project and build them as your own sources. If you are taking this route, these are all the files you'd have to build:
```
├── include
│   └── osdp.h
├── src
│   ├── crypto
│   │   ├── tinyaes.c
│   │   ├── tinyaes_src.c
│   │   └── tinyaes_src.h
│   ├── osdp_common.c
│   ├── osdp_common.h
│   ├── osdp_config.h   <------ This file must be generated
│   ├── osdp_cp.c
│   ├── osdp_export.h   <------ This file must be generated
│   ├── osdp_file.c
│   ├── osdp_file.h
│   ├── osdp_pd.c
│   ├── osdp_phy.c
│   └── osdp_sc.c
└── utils
    ├── include
    │   └── utils
    │       ├── assert.h
    │       ├── disjoint_set.h
    │       ├── list.h
    │       ├── logger.h
    │       ├── pcap_gen.h
    │       ├── queue.h
    │       ├── slab.h
    │       └── utils.h
    └── src
        ├── disjoint_set.c
        ├── list.c
        ├── logger.c
        ├── pcap_gen.c
        ├── queue.c
        ├── slab.c
        └── utils.c
```
Some change might be needed to `utils/` (especially `utils/src/utils.c`) as it has a lot of platform specific code.
